processing/src/main/java/io/druid/query/Query.java

```diff
@@ -70,8 +70,18 @@
 
   String getType();
 
+  /**
+   * @deprecated use {@link QueryPlus#run(QuerySegmentWalker, Map)} instead. This method could be removed in the next
```

Comment:
What new API will they move to if they don't extend `BaseQuery`? Would you plan to add another method to `Query` in a later patch?